commit d4e506a56458384e79ea30f84899425f8ccd625c
Author: maximilian attems <max@stro.at>
Date:   Wed Dec 12 00:42:10 2007 +0100

    [klibc] Kbuild.install header install target cleanup
    
    - only create a dir if we really copy stuff to it.
    - add dual arch asm target (seen missing due to latest x86 arch merge)
    - headers_install in linux-2.6 exports no scsi headers anymore
    
    Cc: Sam Ravnborg <sam@ravnborg.org>
    Signed-off-by: maximilian attems <max@stro.at>

diff --git a/scripts/Kbuild.install b/scripts/Kbuild.install
index 1575169..dd973b1 100644
--- a/scripts/Kbuild.install
+++ b/scripts/Kbuild.install
@@ -95,17 +95,18 @@ header:
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
-	$(Q)set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
-	  mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)include/$$d ; \
+	$(Q)set -e ; for d in linux asm asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
 	  for r in $(KLIBCKERNELSRC)/include $(KLIBCKERNELOBJ)/include \
 	           $(KLIBCKERNELOBJ)/include2 ; do \
-	    [ ! -d $$r/$$d ] || \
-	      cp -rfL $$r/$$d/. \
+	    [ ! -d $$r/$$d ] && continue; \
+	    mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)include/$$d ; \
+	    cp -rfL $$r/$$d/. \
 	          $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/$$d/. ; \
 	  done ; \
 	done
 	$(Q)cd $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include && ln -sf asm-$(KLIBCARCH) asm
 	$(Q)cp -rf usr/include/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/.
+	$(Q)rm -f $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/klibc/.gitignore
 	$(Q)$(install-data) $(srctree)/klcc/klcc.1 $(INSTALLROOT)$(mandir)/man1/$(KCROSS)klcc.1
 	$(Q)$(install-bin) $(objtree)/klcc/$(KCROSS)klcc $(INSTALLROOT)$(bindir)
 
